src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToCopyLocalDependencies.cs

```diff
@@ -258,18 +258,17 @@ public void It_does_not_copy_local_package_dependencies_for_netstandard_projects
         }
 
         [Fact]
-        public void It_copies_local_package_dependencies_for_netstandard_projects_when_requested_to()
+        public void It_does_not_copy_local_runtime_dependencies_for_netframework_projects()
```

Comment:
Should we be concerned about this potentially being a breaking change?  I wouldn't be surprised if some folks were relying on this behavior.

Reply:
Which specifically?  This was previously testing copying assets for netstandard targeted executable projects.  Is that even supported?  In my experiments, there's no way to run them.

Reply:
Erm, never mind, `IsExe` defaults to false in `TestProject`, so this was testing for netstandard libs, my mistake.

Should I instead limit the output of the runtime target assets to `!= ".NETFramework"`?

Reply:
Even with library projects, I think this scenario is probably not supported and won't work in general...

But I bet it works for some folks for their projects, so it would be nice not to break them.

So yes, I think changing it to `!= ".NETFramework"` would be better.

Reply:
Ok I'll amend and restore the test, adding this one as an additional test.

Reply:
I think it's worth taking a step back here.

AFAICT, in 2.x, the behavior would be this for both .NET Standard, .NET Framework with RID == ''

CopyLocalLockFileAssemblies (default for .NET Framework, not default for .NET Standard) -> no runtimes/
Publish -> yes runtimes/

Someone could be depending on this in either case. If we're getting cold feet on one of these, I would get it on both. 

So then I'd go back to having the task return the items and making a call in the targets. That call could be behind an escape hatch.

Reply:
I figured that I had added this test back in the 1x days, and that there was probably a scenario for it back then.  I have a vague notion that people were trying to create .NET Standard test projects, or .NET Standard Exes.  Looking at the history I see that Peter actually added this test as part of his copy local work.  In that case I'm less concerned about the potential breakage, unless he was paterning this off of another test that already existed.

Reply:
Hah, I had forgotten I added this test.

I think I just added it as a check to see if `CopyLocalLockfileAssemblies` was being honored, without thinking through the scenario of actually wanted to copy local assets for `netstandard` TFMs.

Given that we've told people to just [set CopyLocalLockFileAssemblies for netstandard libs](https://github.com/dotnet/cli/issues/6261) in the past, perhaps we shouldn't break anyone relying on this.

I'm in favor of suppressing the task output when the TFM is `.NETFramework`.  I don't see the benefit of always outputting the items and then filtering it based on `'$(TargetFrameworkIdentifier)' != '.NETFramework'` in the targets, but perhaps there is one I'm missing.

Reply:
That jogged my memory that at one time roslyn had their build task as netstandard and were publishing it and running code to probe in the runtimes/. This caused a bunch of problems and they now multi-target. But it does suggest that in rare cases somebody else may be relying on this.

How about we have a boolean that defaults to .NET Core App only but is user-settable and pass that in to the task instead of checking the TFM in the task? Then if someone is doing something exotic, they can set the boolean as an escape hatch. It won't be materially more complicated in code than what we have in this PR, and I would argue it's a saner default to only do .NETCoreApp.

Reply:
Question: is publish still getting runtimes/ to copy from here or still from a different code path?

Reply:
Looks like it depends on this:

https://github.com/dotnet/sdk/blob/98d209f37fdf80760db1595f6611b5835ec5e595/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L298

which would change publish to this path for certain cases:

https://github.com/dotnet/sdk/blob/51aba9eaa1b98c313df7a77bc108846cfb7ba51d/src/Tasks/Microsoft.NET.Build.Tasks/AssetsFileResolver.cs#L52

It would be good to sync these up behind the same flag / tfm check

Reply:
That all still exists because currently publish assets can different from build assets for, primarily, two reasons: packages were marked as excluded from publish or we filtered the packages based on a 2.0 runtime store.

I would dearly love to rip all that out (I have a bug assigned to me to track that), but it would mean that excluded from publish packages no longer end up in the build's deps.json / copied local and that we can no longer support runtime stores.

At any rate, I think passing in a flag to request runtime target assets for `ResolvePackageAssets` makes sense.

Reply:
I think that same flag should be respected in the cases where publish goes through the legacy path.